packages/block-library/src/gallery/edit.js

```diff
@@ -71,146 +65,152 @@ const MOBILE_CONTROL_PROPS_RANGE_CONTROL = Platform.select( {
 
 function GalleryEdit( props ) {
 	const {
+		setAttributes,
 		attributes,
+		clientId,
+		noticeOperations,
 		isSelected,
 		noticeUI,
-		noticeOperations,
-		mediaUpload,
-		imageSizes,
-		resizedImages,
-		onFocus,
+		insertBlocksAfter,
 	} = props;
+
 	const {
-		columns = defaultColumnsNumber( attributes ),
-		imageCrop,
-		images,
+		imageCount,
+		linkTarget,
 		linkTo,
+		columns = defaultColumnsNumber( imageCount ),
 		sizeSlug,
+		imageUploads,
+		imageCrop,
 	} = attributes;
-	const [ selectedImage, setSelectedImage ] = useState();
-	const [ attachmentCaptions, setAttachmentCaptions ] = useState();
+
 	const { __unstableMarkNextChangeAsNotPersistent } = useDispatch(
 		'core/block-editor'
 	);
 
-	function setAttributes( newAttrs ) {
-		if ( newAttrs.ids ) {
-			throw new Error(
-				'The "ids" attribute should not be changed directly. It is managed automatically when "images" attribute changes'
-			);
-		}
+	const currentImageOptions = { linkTarget, linkTo, sizeSlug };
```

Comment:
Every `useEffect` where the `currentImageOptions` is used as a dependency, is called per each render because this is a new reference every time. WDYT about `useMemo` here?

```suggestion
	const currentImageOptions = useMemo(
		() => ( {
			linkTarget,
			linkTo,
			sizeSlug,
		} ),
		[ linkTarget, linkTo, sizeSlug ]
	);
```

Reply:
Thanks @dratwas we just started looking at optimising the useEffects yesterday, so this is timely, we will definitely take a look at adding useMemo where appropriate.